functions/src/modules/gmail.functions.ts

```diff
@@ -0,0 +1,273 @@
+      decodedMessage.emailAddress
+    );
+    oauth2Client.setCredentials({
+      access_token: oauth.token,
+      refresh_token: oauth.refreshToken,
+    });
+    const gm = await Google.google.gmail({
+      version: 'v1',
+      auth: oauth2Client,
+    });
+
+    const labelResults = await gm.users.labels.list({
+      userId: 'me',
+    });
+    const doneLabelObj = labelResults.data.labels.filter(label => {
+      return label.name === 'SQRDone';
+    })[0];
+
+    try {
+      // Typescript complains if not initialized and casted to any
+      const historyListRequest: any = {
+        userId: decodedMessage.emailAddress || 'me',
+        startHistoryId: lastSyncHistoryId,
+        historyTypes: 'labelAdded',
+        labelId: [doneLabelObj.id],
+      };
+
+      // Call the history list api to fetch changes since last synced
+      const results = await gm.users.history.list(historyListRequest);
+      if (!results.data.history || !results.data.history.length) {
+        // Handle empty history, and update history id
+        await storeHistoryIdInDatabase(
+          decodedMessage.emailAddress,
+          decodedMessage.historyId
+        );
+        throw new Error('History is empty. Nothing to process');
+      }
+
+      const onlyLabelAdded = results.data.history.filter(change => {
+        return change.labelsAdded;
+      });
+
+      // We only care about the final state of the Done label whether it was applied
+      // or not, so we get rid of the duplicates incase there are any
+      const uniques = {};
+      onlyLabelAdded.forEach(change => {
+        change.labelsAdded.forEach(obj => {
+          uniques[obj.message.id] = {
+            labels: obj.labelIds,
+            threadId: obj.message.threadId,
+          };
+        });
+      });
+
+      // Call the gmail.users.threads api with the unique changes
+      // and retrieve list of threads and messages
+      const threadHeaders = Object.keys(uniques).map(async threadKey => {
+        const threadResults = await gm.users.threads.get({
+          userId: 'me',
+          id: uniques[threadKey].threadId,
+        });
+        // Only care about the first message headers in the thread
+        return threadResults.data.messages[0].payload.headers;
+      });
+      const allResults = await Promise.all(threadHeaders);
+      // Filter the subject header out from the response
+      const subjectParts = allResults.map(thread => {
+        const subjectValue = thread.filter(
+          header => header.name === 'Subject'
+        )[0].value;
+        if (subjectValue) {
+          const p = subjectValue.split(' - ');
+          return { no: p[0], fileName: p[1], person: p[2] };
+        }
+        return null;
+      });
+
+      subjectParts.forEach(async subjectPart => {
+        const list = subjectPart.fileName.split('-')[0];
+        const Ref = db.ref(
+          `files/${list}/${subjectPart.fileName}/soundQualityReporting`
```

Comment:
Audio file name should be obtained from the `X-Audio-Seva-File-Name` header of the message. Added this to the card description.

Reply:
Now you have a TEST email in the mailbox with this header.